### ISSUE TITLE
Leaking goroutine on Server.Run

### ISSUE THREAD

**Issue by paskal** [go-pkgz/jrpc#4]
With following change in place I see report on leaking goroutines:

```diff
diff --git a/server_test.go b/server_test.go
index 30c07d5..851c09a 100644
--- a/server_test.go
+++ b/server_test.go
@@ -12,9 +12,11 @@ import (
 	"github.com/pkg/errors"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
 )
 
 func TestServerPrimitiveTypes(t *testing.T) {
+	defer goleak.VerifyNone(t)
 	s := Server{API: "/v1/cmd", Logger: NoOpLogger}
 
 	type respData struct {
```

Report:

```go
=== RUN   TestServerPrimitiveTypes
--- FAIL: TestServerPrimitiveTypes (0.46s)
    server_test.go:32: [blah 42 true]
    server_test.go:32: [blah 42 true]
    leaks.go:78: found unexpected goroutines:
        [Goroutine 37 in state select, with github.com/patrickmn/go-cache.(*janitor).Run on top of the stack:
        goroutine 37 [select]:
        github.com/patrickmn/go-cache.(*janitor).Run(0xc00011c3c0, 0xc000132880)
        	/Users/dmitry/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1079 +0xd5
        created by github.com/patrickmn/go-cache.runJanitor
        	/Users/dmitry/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1099 +0xa9
        
         Goroutine 38 in state select, with github.com/patrickmn/go-cache.(*janitor).Run on top of the stack:
        goroutine 38 [select]:
        github.com/patrickmn/go-cache.(*janitor).Run(0xc00011c3d0, 0xc0001328c0)
        	/Users/dmitry/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1079 +0xd5
        created by github.com/patrickmn/go-cache.runJanitor
        	/Users/dmitry/go/pkg/mod/github.com/patrickmn/go-cache@v2.1.0+incompatible/cache.go:1099 +0xa9
        ]
```

I haven't yet figured out what is the root cause and or how to fix it.

The effect of this leak is unknown to me, I just know that's it's visible with [goleak](https://github.com/uber-go/goleak) tool both in this project and also in remark42 which uses it. Feel free to close if you don't think it's worth investigating.

**Comment by umputun** [comment]
it looks like the janitor thread from go-cache, which is a transitive dependency of github.com/didip/tollbooth (limiter)

```
go mod why github.com/patrickmn/go-cache
# github.com/patrickmn/go-cache
github.com/go-pkgz/jrpc
github.com/didip/tollbooth
github.com/didip/tollbooth/limiter
github.com/patrickmn/go-cache
```

**Comment by paskal** [comment]
I've tested tollbooth itself and it doesn't leak goroutines. It seems that here in `jrpc` something is not right.

**Comment by paskal** [comment]
Should be fixed by 794c304098e1f101a3611eb600c4d6701718d768, I'll check again shortly. I wonder why I didn't update that issue right away back then.

**Comment by paskal** [comment]
Verified, no leaks in the current master (v0.2.0). Resolving.
